presto-docs/src/main/sphinx/connector/oracle.rst

```diff
@@ -23,47 +20,353 @@ connection properties as appropriate for your setup:
+    - See :ref:`number mapping`
+  * - ``FLOAT[(p)]``
+    - ``DOUBLE``
+    -
+  * - ``BINARY_FLOAT``
+    - ``REAL``
+    -
+  * - ``BINARY_DOUBLE``
+    - ``DOUBLE``
+    -
+  * - ``VARCHAR2(n CHAR)``
+    - ``VARCHAR(n)``
+    -
+  * - ``VARCHAR2(n BYTE)``
+    - ``VARCHAR(n)``
+    -
+  * - ``NVARCHAR2(n)``
+    - ``VARCHAR(n)``
+    -
+  * - ``CHAR(n)``
+    - ``CHAR(n)``
+    -
+  * - ``NCHAR(n)``
+    - ``CHAR(n)``
+    -
+  * - ``CLOB``
+    - ``VARCHAR``
+    -
+  * - ``NCLOB``
+    - ``VARCHAR``
+    -
+  * - ``RAW(n)``
+    - ``VARBINARY``
+    -
+  * - ``BLOB``
+    - ``VARBINARY``
+    -
+  * - ``DATE``
+    - ``TIMESTAMP``
+    - See :ref:`datetime mapping`
+  * - ``TIMESTAMP(p)``
+    - ``TIMESTAMP``
+    - See :ref:`datetime mapping`
+  * - ``TIMESTAMP(p) WITH TIME ZONE``
+    - ``TIMESTAMP WITH TIME ZONE``
+    - See :ref:`datetime mapping`
+
+If an Oracle table contains a type that's not listed in the above table, an
+error shows:
+
+ - An ``unsupported-type.handling-strategy`` equals ``FAIL``
+   (default configuration), any queries on the table fail.
+ - An ``unsupported-type.handling-strategy`` equals ``IGNORE``
+   (default configuration), Presto no longer accesses the column.
+ - An ``unsupported-type.handling-strategy`` equals ``CONVERT_TO_VARCHAR``
+   (default configuration), a column value converts to an unbounded ``VARCHAR``.
+
+Presto to Oracle type mapping
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Presto supports creating tables with the following types in an Oracle database.
+The table shows the mappings from Presto to Oracle data types:
+
+.. note::
+   For types not listed in the table below, Presto can't perform the ``CREATE
+   TABLE <table> AS SELECT`` operations. When data is inserted into existing
+   tables ``Oracle to Presto`` type mapping is used.
+
+.. list-table:: Presto to Oracle Type Mapping
+  :widths: 20, 20, 60
+  :header-rows: 1
+
+  * - Presto type
+    - Oracle database type
+    - Notes
+  * - ``TINYINT``
+    - ``NUMBER(3)``
+    -
+  * - ``SMALLINT``
+    - ``NUMBER(5)``
+    -
+  * - ``INTEGER``
+    - ``NUMBER(10)``
+    -
+  * - ``BIGINT``
+    - ``NUMBER(19)``
+    -
+  * - ``DECIMAL(p, s)``
+    - ``NUMBER(p, s)``
+    -
+  * - ``REAL``
+    - ``BINARY_FLOAT``
+    -
+  * - ``DOUBLE``
+    - ``BINARY_DOUBLE``
+    -
+  * - ``VARCHAR``
+    - ``NCLOB``
+    -
+  * - ``VARCHAR(n)``
+    - ``VARCHAR2(n CHAR)`` or ``NCLOB``
+    - See :ref:`character mapping`
+  * - ``CHAR(n)``
+    - ``CHAR(n CHAR)`` or ``NCLOB``
+    - See :ref:`character mapping`
+  * - ``VARBINARY``
+    - ``BLOB``
+    -
+  * - ``DATE``
+    - ``DATE``
+    - See :ref:`datetime mapping`
+  * - ``TIMESTAMP``
+    - ``TIMESTAMP(3)``
+    - See :ref:`datetime mapping`
+  * - ``TIMESTAMP WITH TIME ZONE``
+    - ``TIMESTAMP(3) WITH TIME ZONE``
+    - See :ref:`datetime mapping`
+
+.. _number mapping:
+
+Mapping numeric types
+^^^^^^^^^^^^^^^^^^^^^
+
+An Oracle ``NUMBER(p, s)`` maps to Presto's ``DECIMAL(p, s)`` except in these
```

Comment:
This section renders incorrectly:

![image](https://user-images.githubusercontent.com/890111/92028881-0899d300-ed65-11ea-85f0-740616efc284.png)

Reply:
Fixed